compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -871,6 +873,8 @@ static LogicalResult setMatmulNoPadRootConfig(
             std::back_inserter(newTileSizes));
   newTileSizes.push_back(parallelTileSizes);
   newTileSizes.push_back(reductionTileSizes);
+  // No need for tiling inner parallel dims.
+  newTileSizes.emplace_back(parallelTileSizes.size(), 0);
```

Comment:
Wondering if we really have to enforce a 4th level for all the ops given that the last one is not applicable to most of them. We can have a 4th level only for the ops that need it. `TilingConfig` should be able to handle configurations with 3, 4 and 5 levels and do the proper verification to make sure that each configuration is valid. WDYT?